sdk/src/main/java/io/dapr/client/DaprHttp.java

```diff
@@ -174,12 +200,16 @@
    * @param json Response body from Dapr.
    * @return DaprError or null if could not parse.
    */
-  private static DaprError parseDaprError(byte[] json) throws IOException {
+  private static DaprError parseDaprError(String json) {
     if (json == null) {
       return null;
     }
 
-    return OBJECT_MAPPER.readValue(json, DaprError.class);
+    try {
+      return OBJECT_MAPPER.readValue(json, DaprError.class);
+    } catch (IOException e) {
+      throw new DaprException("500", "Unknown error: could not parse error json.");
```

Comment:
DaprException should be only for exceptions sent from Dapr's runtime.  Plus, the error code string does not map to HTTP error code. It is custom strings defined in the runtime code in Go.